### PR TITLE
Framework Upgrades, RN 0.69.4, Expo 46

### PR DESCRIPTION
**Pull request by frankcalise**
## Please verify the following:

- [x] `yarn ci:test` **jest** tests pass with new tests, if relevant
- [x] `README.md` has been updated with your changes, if relevant

## Describe your PR

Implementation for #1979 

- Expo 46
- React Native 0.69.4
- React 18
- Removed validate.js and utils/validate.ts
- `yarn expo:web` has issues running via Expo's new local CLI (can still run via `expo-cli expo start --web`

### CODE REVIEW

**Comment by frankcalise** [comment]
WIP 🔧 

*Working*
- yarn ios
- yarn expo:start

*In Progress*
- yarn android
-yarn expo:web

**Comment by frankcalise** [comment]
I think Expo web currently has an issue with the new local CLI (yarn expo:web or yarn expo start --web). If you start up the app with the old global CLI that they're migrating from, it does launch (expo-cli expo start --web)

**Review by jamonholmgren — APPROVED** [review]

File: boilerplate/android/app/src/main/java/com/helloworld/MainActivity.java [hunk 1]
```diff
@@ -28,6 +30,12 @@ protected ReactActivityDelegate createReactActivityDelegate() {
     return new ReactActivityDelegateWrapper(this, new MainActivityDelegate(this, getMainComponentName()));
   }
 
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    RNBootSplash.init(this);            // <- initialize the splash screen
+    super.onCreate(savedInstanceState); // or super.onCreate(null) with react-native-screens
```
Comment (yulolimum) [inline boilerplate/android/app/src/main/java/com/helloworld/MainActivity.java:36]: @frankcalise do we wanna do null here?

**Review by yulolimum — APPROVED** [review]

File: boilerplate/android/app/src/main/java/com/helloworld/MainActivity.java (hunk 1 above)
Comment (frankcalise) [reply boilerplate/android/app/src/main/java/com/helloworld/MainActivity.java:36]: @yulolimum yeah my bad I had did it locally in a test project - thanks!

**Pull request merged**